### PR TITLE
[BAC-1985] - Refactor and upgrade project dependencies

### PR DESCRIPTION
**Pull request by didasmbalanya**
Line by line update

Commits will be squashed once updates are complete

### CODE REVIEW

File: app/fileReceiver.js [hunk 1]
```diff
@@ -70,15 +70,15 @@ export default class FileReceiver extends Nanobus {
     this.state = 'ready';
   }
 
-  async reportLink(reason) {
-    await reportLink(this.fileInfo.id, this.keychain, reason);
-  }
+  // async reportLink(reason) {
+  //   await reportLink(this.fileInfo.id, this.keychain, reason);
+  // }
```
Comment (alsprogrammer) [inline app/fileReceiver.js]: Is it supposed to be used in future?

File: server/routes/download.js [hunk 2]
```diff
@@ -2,20 +2,72 @@ const storage = require('../storage');
...
+      // either not available at all or expired already...
+      return res.sendStatus(404);
+    }
+
+    const contentType = meta.contentType || 'application/octet-stream';
+
+    const length = await storage.length(id);
+    if (!length) {
+      log.warn('length', {
+        msg: 'Stale metadata (file doe not exist)',
```
Comment (alsprogrammer) [inline server/routes/download.js]: `Stale metadata (file does not exist)`, right?

**Review by alsprogrammer — APPROVED** [review]
LGTM!

File: app/fileReceiver.js (hunk 1 above)
Comment (ghost) [reply app/fileReceiver.js]: Good point, it's dead code. Removed it.

File: server/routes/download.js (hunk 2 above)
Comment (ghost) [reply server/routes/download.js]: Thx for spotting this. FIxed.

**Pull request merged**